### ISSUE TITLE
The current state of ESI

### ISSUE THREAD

**Issue by ccee2abf-fb28-497b-88ca-ae3643da4a8f** [esi/esi-issues#1225]
See title. 

Proposed changes such as the new bookmark endpoint and even smaller changes such as new notifications in the notification endpoint appear to be either heavily delayed or just not happening at all. 

If ESI is just being maintained to keep it going, and no new features are considered. Please consider sharing that information with developers so that they can manage expectations.

**Comment by alexbaileyuk** [comment]
Agree. ESI is a great feature for developers and gives players loads of benefit. Right now it feels like CCP is giving a very CBA approach.

I understand that priorities are driven by other things but managing expectations is important.

**Comment by HakShak** [comment]
Here's the "short" version:

A lot of work has gone into moving ESI endpoints internally from JSON to Protobuf. It's a long journey:

![image](https://user-images.githubusercontent.com/4952085/89675761-049ba200-d8da-11ea-9fcc-8d2df24c5faf.png)

This is slow because we are using ESI to properly model large portions of Eve Online through [Protobuf](https://developers.google.com/protocol-buffers). This means undoing/untangling/redefining designs which have not aged well and in some cases are incompatible with the patterns required for systems like ESI.

Furthermore, we have had to pause this progress as we support teams who are learning and using the technology born out of ESI. The latest of which being the [Proving Grounds Leaderboards](https://www.eveonline.com/article/qcp33s/proving-of-kybernauts-for-glorification-in-the-flow-unfolds).

On top of that a lot of support goes to the teams modelling more and more of the Eve Universe in Protobuf as newer features are added to internal tooling and [Portal](https://www.eveonline.com/eve-portal).

Our current tasks have us focused on two highly coupled tasks: gRPC in the Eve Client and taking local chat out back to be shot. Repeatedly. Until it connects.

Once we emerge from that work we can reevaluate where we are.

**Comment by ddouglas** [comment]
I just want be the first to say Thank You for not ignoring this and providing the update. We will most likely be using this issue as the response to questions of this genre for the foreseeable future.

**Comment by ccee2abf-fb28-497b-88ca-ae3643da4a8f** [comment]
Thanks for being transparent 🥰

**Comment by colcrunch** [comment]
@HakShak any chance of another state of ESI update a year later?

**Comment by HakShak** [comment]
So [that's](https://www.eveonline.com/news/view/introducing-quasar) out of the way we can hopefully speak more to ESI, but right now we would still need to prioritize the migration from above. Quasar might also change the way we expose things in the future.

**Comment by notmorganfreewoman** [comment]
> So [that's](https://www.eveonline.com/news/view/introducing-quasar) out of the way we can hopefully speak more to ESI, but right now we would still need to prioritize the migration from above. Quasar might also change the way we expose things in the future.

Hopefully GraphQL for a public interface as requested here https://github.com/esi/esi-issues/issues/1309

Also if there is a problem with commitment of resources and time from CCP on a public API perhaps they should consider charging a fee to access/use it? Whilst I understand that suggestion would be unpopular, it would certinally be an incentive for CCP to improve things on the API.

**Comment by evanova** [comment]
> > So [that's](https://www.eveonline.com/news/view/introducing-quasar) out of the way we can hopefully speak more to ESI, but right now we would still need to prioritize the migration from above. Quasar might also change the way we expose things in the future.
> 
> Hopefully GraphQL for a public interface as requested here #1309
> 
> Also if there is a problem with commitment of resources and time from CCP on a public API perhaps they should consider charging a fee to access/use it? 

Given that we do not make any actual money for our work, charging a fee for ESI is the worse idea I've read in a long time and most likely the best way to halt third-party's dedication. it makes no sense whatsoever, and is the exact opposite of the current partner's program idea which is for us not having to spend money to maintain our apps.

Now reading this thread, what I still wonder about is why a public API like ESI is tied to the Eve client or anything else other than just being a facility for third-parties. Isn't that a source of a lot of development pain? I imagine having to think about not breaking Eve or other apps while changing some public REST endpoints makes the whole process rather heavy and difficult.
Maybe what could happen is decoupling Eve's internal from the public part of ESI.  This would allow everyone a lot more flexibility in making changes on both sides.

**Comment by notmorganfreewoman** [comment]
> charging a fee for ESI is the worse idea I've read in a long time and most likely the best way to halt third-party's dedication

We already have to have a valid payment registered to get a _devkey_, whilst that makes sense for online web services and websites centralised services, it makes absolutely ZERO sense and a RISK for desktop published applications that are outside the control of the developer. Many developers are now opting to ship with the user bringing (registering their own payment method with CCP to obtain it) their own _devkey_ to use it. Even Evemon now has a field to use your own devkey. All my apps for the desktop will force users to use their own devkey. They won't be using MY devkey that I paid for (registered a payment method for on CCP's site).

I will NEVER allow MY devkey to be used on desktop published apps. EVER. Use your own devkey (and thus pay for it), or build your own.

**Comment by fuzzysteve** [comment]
Just to be clear, having been around and involved with the discussions wrt the payment thing:

The valid payment thing is _only_ so CCP can identify who you are in the real world. It's not so CCP make sure you paid them money. It's why Paypal didn't count for it. Only credit cards. 

Payment for API access or development access is a terrible idea, which has been talked about in the past, and thrown out as a concept. It will stifle development. I know I, for one, would not pay for it. I would walk instead.

Also, please don't try to speak for CCP. You don't work for them. You don't have a pipeline to them behind the scenes.

**Comment by ccp-pinky** [comment]
I think the recent [EVE Evolved posts covering ESI](https://www.eveonline.com/news/view/esi-delivered-the-next-chapter) says it all :) Going to close this ticket now :)
